### DISCUSSION TITLE
[Task]: Implement the 3D Anomaly Detection method BTF

### DISCUSSION THREAD

**Discussion by alexriedel1** [open-edge-platform/anomalib#1262] [Feature Requests]
### What is the motivation for this task?

The [3D MVTec dataset module in anomalib](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/data/mvtec_3d.py)  allows for testing 3D anomaly detection models. A simple yet good performing model on this dataset is called "Back to the Feature (BTF)", introduced here: https://arxiv.org/pdf/2203.05550v3.pdf

It basically just uses the already implemented [PatchCore ](https://github.com/openvinotoolkit/anomalib/tree/main/src/anomalib/models/patchcore) algorithm and samples both extracted 2D image features and [FPFH](https://pcl.readthedocs.io/projects/tutorials/en/latest/fpfh_estimation.html) point cloud features.

There is an [implementation of the model by the paper authors](https://github.com/eliahuhorwitz/3D-ADS) available which should make it easy so implement in anomalib 

### Describe the solution you'd like

1. Read the BTF paper
2. check the authors implementation on GitHub
3. implement the model in anomalib based on the already available PatchCore model
4. match the authors accuracy metrices

**Comment by samet-akcay** [comment]
@alexriedel1, thanks again for the suggestion. I agree that the library should have a model from the 3d anomaly detection benchmark. Thanks to your efforts Anomalib has now MVTec 3d dataset support. Would be great to have a model support too.

**Reply by vmiller987** [reply]
@samet-akcay, Is there a plan for 3D anomaly detection? I haven't been given a 3D anomaly dataset yet, but I was asked to start researching this area. I found this repo which summarizes papers that cover 3D, but it is 2 years old. (https://github.com/wyzjack/Awesome-3D-ADL) This field moves very fast, but I imagine some of these are still good.

I haven't kept up with all the updates to Anomalib, but I'm very impressed with the progress made. Keep up the good work. I would love to stay with Anomalib for all of our anomaly detection.
